code/main/edge/locationHintManager.brs

```diff
@@ -64,24 +60,40 @@ function _adb_LocationHintManager() as object
                     continue for
                 end if
 
-                if not _adb_isEmptyOrInvalidString(payload.scope) and LCase(payload.scope) = m._EDGE_NETWORK_SCOPE
+                if _adb_stringEqualsIgnoreCase(payload.scope, m._EDGE_NETWORK_SCOPE)
                     m.setLocationHint(payload.hint, payload.ttlSeconds)
                 end if
 
             end for
         end function,
 
+        _setExpiryTime: function(ttlSeconds as dynamic) as void
+            if _adb_isInvalidInt(ttlSeconds)
+                _adb_logDebug("_adb_LocationHintManager::_setExpiryTime() - ttlSeconds is not found, using default ttl (" + FormatJson(m._DEFAULT_LOCATION_HINT_TTL_SEC) + ") seconds.")
+                ttlSeconds = m._DEFAULT_LOCATION_HINT_TTL_SEC
+            end if
+
+            m._initTSInMillis = _adb_timestampInMillis()
+            m._expiryTSInMillis = m._initTSInMillis + (ttlSeconds * 1000)
```

Comment:
I'm not sure about the possible values for `ttlSeconds`. If it can be a large integer, could we do an additional check to prevent integer overflow when calculating `m._initTSInMillis + (ttlSeconds * 1000)`.

Reply:
I don't think we would have such a high number. And shouldn't the variable type change accordingly if it does? since we don't assign types?

Reply:
https://developer.roku.com/docs/references/brightscript/language/expressions-variables-types.md#type-conversion-promotion
Yeah, it seems that the type conversion may be applied in that situation.  Never mind, please ignore this comment. 
And if possible, do you mind doing a quick test locally to verify the type conversion?

Reply:
I have refactored code to take input type as longinteger. Let me know what do you think?

Reply:
Yes, it makes sense.